diff --git a/view/tpl/profile_edit.tpl b/view/tpl/profile_edit.tpl
index c967fa02c3702bc564ce0a7003b4a290739988a8..8add9bb9d9aea294a7d1cdc77526a1b90914c4b0 100755
--- a/view/tpl/profile_edit.tpl
+++ b/view/tpl/profile_edit.tpl
@@ -294,7 +294,11 @@
 							{{if $fields.sexual}}
 							<div id="profile-edit-sexual-wrapper" class="form-group field" >
 							<label id="profile-edit-sexual-label" for="sexual-select" >{{$lbl_sexual}}</label>
+							{{if $advanced}}
 							{{$sexual}}
+							{{else}}
+							{{$sexual_min}}
+							{{/if}}
 							</div>
 							<div class="clear"></div>
 							{{/if}}
diff --git a/include/feedutils.php b/include/feedutils.php
index 9ff09cc662ae5ff67c412205ad6930b3ef973f1f..0a9af7ee1faed4ac14c67532b3406737f9dc8245 100644
--- a/include/feedutils.php
+++ b/include/feedutils.php
@@ -452,6 +452,7 @@ function get_atom_elements($feed, $item) {
 	else {
 	    $res['title'] = bbcode($res['title'], [ 'tryoembed' => false ]);
 	    $res['title'] = html2plain($res['title'], 0, true);
+		$res['title'] = strip_tags($res['title']);
 	    $res['title'] = html_entity_decode($res['title'], ENT_QUOTES, 'UTF-8');
 	    $res['title'] = preg_replace("/https?\:\/\/[a-zA-Z0-9\:\/\-\?\&\;\.\=\_\~\#\%\$\!\+\,\@]+/", "", $res['title']);
 	    while (strpos($res['title'], "\n") !== false)
diff --git a/Zotlabs/Module/Item.php b/Zotlabs/Module/Item.php
index fcc040e017a4fd31114bee12b6556b93a4bf7825..cdfc3c9a84c94feb7974062ea95f62f78810649b 100644
--- a/Zotlabs/Module/Item.php
+++ b/Zotlabs/Module/Item.php
@@ -723,16 +723,16 @@ class Item extends Controller {
 			// we may need virtual or template classes to implement the possible alternatives
 
 			if(strpos($body,'[/summary]') !== false) {
-                $match = '';
-                $cnt = preg_match("/\[summary\](.*?)\[\/summary\]/ism",$body,$match);
-                if($cnt) {
-                    $summary .= $match[1];
-                }
-                $body_content = preg_replace("/^(.*?)\[summary\](.*?)\[\/summary\](.*?)$/ism", '',$body);
-                $body = trim($body_content);
-            }
-
-            $summary = cleanup_bbcode($summary);
+			    $match = '';
+			    $cnt = preg_match("/\[summary\](.*?)\[\/summary\]/ism",$body,$match);
+			    if($cnt) {
+			        $summary .= $match[1];
+			    }
+			    $body_content = preg_replace("/\[summary\](.*?)\[\/summary\]/ism", '',$body);
+			    $body = trim($body_content);
+			}
+			
+			$summary = cleanup_bbcode($summary);
 
 			$body = cleanup_bbcode($body);
