diff --git a/Zotlabs/Lib/System.php b/Zotlabs/Lib/System.php
index 7bf1343bb..7d2975f0c 100644
--- a/Zotlabs/Lib/System.php
+++ b/Zotlabs/Lib/System.php
@@ -5,9 +5,14 @@ namespace Zotlabs\Lib;
 class System {
 
 	static public function get_platform_name() {
-		if(is_array(\App::$config) && is_array(\App::$config['system']) && array_key_exists('platform_name',\App::$config['system']))
-			return \App::$config['system']['platform_name'];
-		return PLATFORM_NAME;
+		static $platform_name = PLATFORM_NAME;
+		static $is_platform_name_set = false;
+		if (! $is_platform_name_set) {
+			if(is_array(\App::$config) && is_array(\App::$config['system']) && array_key_exists('platform_name',\App::$config['system']))
+				$platform_name = \App::$config['system']['platform_name'];
+			$is_platform_name_set = true;
+		}
+		return $platform_name;
 	}
 
 	static public function get_site_name() {
diff --git a/include/items.php b/include/items.php
index 917808ad5..74ceff6aa 100755
--- a/include/items.php
+++ b/include/items.php
@@ -4066,11 +4066,11 @@ function fetch_post_tags($items, $link = false) {
 		);
 	}
 
-	for($x = 0; $x < count($items); $x ++) {
-		if($tags) {
-			foreach($tags as $t) {
-				if(($link) && ($t['ttype'] == TERM_MENTION))
-					$t['url'] = chanlink_url($t['url']);
+	if($tags) {
+		foreach($tags as $t) {
+			if(($link) && ($t['ttype'] == TERM_MENTION))
+				$t['url'] = chanlink_url($t['url']);
+			for($x = 0; $x < count($items); $x ++) {
 				if(array_key_exists('item_id',$items[$x])) {
 					if($t['oid'] == $items[$x]['item_id']) {
 						if(! is_array($items[$x]['term']))
@@ -4087,9 +4087,11 @@ function fetch_post_tags($items, $link = false) {
 				}
 			}
 		}
-		if($imeta) {
-			foreach($imeta as $i) {
-				if(array_key_exists('item_id',$items[$x])) {
+	}
+	if($imeta) {
+		for($x = 0; $x < count($items); $x ++) {
+			if(array_key_exists('item_id',$items[$x])) {
+				foreach($imeta as $i) {
 					if($i['iid'] == $items[$x]['item_id']) {
 						if(! is_array($items[$x]['iconfig']))
 							$items[$x]['iconfig'] = array();
@@ -4097,7 +4099,9 @@ function fetch_post_tags($items, $link = false) {
 						$items[$x]['iconfig'][] = $i;
 					}
 				}
-				else {
+			}
+			else {
+				foreach($imeta as $i) {
 					if($i['iid'] == $items[$x]['id']) {
 						if(! is_array($items[$x]['iconfig']))
 							$items[$x]['iconfig'] = array();
diff --git a/include/language.php b/include/language.php
index e9d62e434..a2c717308 100644
--- a/include/language.php
+++ b/include/language.php
@@ -217,9 +217,9 @@ function t($s, $ctx = '') {
  */
 
 function translate_projectname($s) {
-
-	return str_replace(array('$projectname','$Projectname'),array(Zotlabs\Lib\System::get_platform_name(),ucfirst(Zotlabs\Lib\System::get_platform_name())),$s);
-
+	if(strpos($s,'rojectname') !== false)
+		return str_replace(array('$projectname','$Projectname'),array(Zotlabs\Lib\System::get_platform_name(),ucfirst(Zotlabs\Lib\System::get_platform_name())),$s);
+	return $s;
 }
 
 
